### PR TITLE
[graphql/rpc] add move package

### PR DESCRIPTION
**Pull request by oxade**
## Description 

Adds MovePackage type.
~~Seems DB is broken so currently using RPC 1.0~~
More examples and MoveModule will follow in subs PRs as the results are bland otherwise

## Test Plan 

Manual

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/src/types/move_module.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::*;
+use move_binary_format::CompiledModule;
+
+#[derive(Clone)]
+pub(crate) struct MoveModule {
+    pub native_module: CompiledModule,
```
[the remark this answers is not in the document]
Comment (wlmyng) [reply crates/sui-graphql-rpc/src/types/move_module.rs:9]: it'd save us from having to include move-binary-format as a dependency

**Comment by oxade** [comment]
> suggestion: rewrite this as `impl From<SuiRawMovePackage> for MovePackage`? does this need to be Result<T>, don't see any fallible conversions 

Wont work because of the additional `max_move_package_size`

**Comment by oxade** [comment]
> would NormalizedMoveModule or sui-json-rpc-types::SuiMoveNormalizedModule be easier to work with 

I wouldn't want to couple RPC 2.0 to RPC 1.0 in that manner

**Review by wlmyng — APPROVED** [review]

File: crates/sui-graphql-rpc/src/types/object.rs [hunk 2]
```diff
@@ -94,6 +95,30 @@ impl Object {
         self.owner.as_ref().map(|q| Owner { address: *q })
     }
 
+    async fn as_move_package(&self) -> Result<Option<MovePackage>> {
+        if let Some(bcs) = &self.bcs {
+            let bytes = bcs.0.as_slice();
+
+            // Need to deser using `sui_json_rpc_types::SuiRawMovePackage` because it uses custom serde
+            let package = bcs::from_bytes::<sui_json_rpc_types::SuiRawMovePackage>(bytes)
```
[the remark this answers is not in the document]
Comment (oxade) [reply crates/sui-graphql-rpc/src/types/object.rs:103]: > wouldnt these be Data::Package and Data::Move given how we serialize them here https://github.com/MystenLabs/sui/blob/main/crates/sui-graphql-rpc/src/context_data/db_data_provider.rs#L1152

It should be, but I was using sui json RPC 1.0 here due to DB not being available. Is the DB available now?

**Pull request merged**